File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.java
class Solution {
    private void traverse(Node node, List<Integer> result) {
        if (node == null) return;
        for (Node child: node.children) {
            traverse(child, result);
        }
        result.add(node.val);
    }

    public List<Integer> postorder(Node root) {
        List<Integer> result = new ArrayList<>();
        traverse(root, result);
        return result;
    }
}

File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.cpp
class Solution {
    void traverse(Node* node, vector<int>& result) {
        if (node == NULL) return;
        for (int i = 0; i < node->children.size(); i++) {
            traverse(node->children[i], result);
        }
        result.push_back(node->val);
    }

public:
    vector<int> postorder(Node* root) {
        vector<int> result;
        traverse(root, result);
        return result;
    }
};